clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -335,10 +336,17 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
                 throw new IllegalStateException("Consumer is not subscribed to any topics or assigned any partitions");
             }
 
+            final CompletableFuture<Void> wakeupFuture = setupWakeupTrigger();
```

Comment:
I needed to create a dedicated future since fetches are not passed to the background thread as events since fetches happen continuously on the background thread.